proto/internal/uber/cadence/admin/v1/service.proto

```diff
@@ -0,0 +1,286 @@
+
+  // CloseShard closes shard.
+  rpc CloseShard(CloseShardRequest) returns (CloseShardResponse);
+
+  // RemoveTask removes task.
+  rpc RemoveTask(RemoveTaskRequest) returns (RemoveTaskResponse);
+
+  // ResetQueue resets queue.
+  rpc ResetQueue(ResetQueueRequest) returns (ResetQueueResponse);
+
+  // DescribeQueue describes queue.
+  rpc DescribeQueue(DescribeQueueRequest) returns(DescribeQueueResponse);
+
+  // Returns the raw history of specified workflow execution.
+  // It fails with 'EntityNotExistError' if specified workflow execution in unknown to the service.
+  // StartEventId defines the beginning of the event to fetch. The first event is inclusive.
+  // EndEventId and EndEventVersion defines the end of the event to fetch. The end event is exclusive.
+  rpc GetWorkflowExecutionRawHistoryV2(GetWorkflowExecutionRawHistoryV2Request) returns(GetWorkflowExecutionRawHistoryV2Response);
+
+  // GetReplicationMessages returns new replication tasks since the read level provided in the token.
+  rpc GetReplicationMessages(GetReplicationMessagesRequest) returns(GetReplicationMessagesResponse);
+
+  // GetDLQReplicationMessages return replication messages based on DLQ info.
+  rpc GetDLQReplicationMessages(GetDLQReplicationMessagesRequest) returns (GetDLQReplicationMessagesResponse);
+
+  // GetDomainReplicationMessages returns new domain replication tasks since last retrieved task id.
+  rpc GetDomainReplicationMessages (GetDomainReplicationMessagesRequest) returns (GetDomainReplicationMessagesResponse);
+
+  // ReapplyEvents applies stale events to the current workflow and current run.
+  rpc ReapplyEvents(ReapplyEventsRequest) returns (ReapplyEventsResponse);
+
+  // AddSearchAttribute whitelist search attribute in request.
+  rpc AddSearchAttribute (AddSearchAttributeRequest) returns (AddSearchAttributeResponse);
+
+  // DescribeCluster returns information about Cadence cluster.
+  rpc DescribeCluster(DescribeClusterRequest) returns (DescribeClusterResponse);
+
+  // ReadDLQMessages returns messages from DLQ.
+  rpc ReadDLQMessages(ReadDLQMessagesRequest) returns (ReadDLQMessagesResponse);
+
+  // PurgeDLQMessages purges messages from DLQ.
+  rpc PurgeDLQMessages(PurgeDLQMessagesRequest) returns (PurgeDLQMessagesResponse);
+
+  // MergeDLQMessages merges messages from DLQ.
+  rpc MergeDLQMessages(MergeDLQMessagesRequest) returns (MergeDLQMessagesResponse);
+
+  // RefreshWorkflowTasks refreshes all tasks of a workflow.
+  rpc RefreshWorkflowTasks(RefreshWorkflowTasksRequest) returns (RefreshWorkflowTasksResponse);
+
+  // ResendReplicationTasks requests replication tasks from remote cluster and apply tasks to current cluster.
+  rpc ResendReplicationTasks(ResendReplicationTasksRequest) returns (ResendReplicationTasksResponse);
+}
+
+message DescribeWorkflowExecutionRequest {
+  string domain = 1;
+  api.v1.WorkflowExecution workflow_execution = 2;
+}
+
+message DescribeWorkflowExecutionResponse {
+  string shard_id = 1;
```

Comment:
Why is shard_id a string instead of an int32?

Reply:
This was missed, as it was ported from Thrift where this is weirdly a string: https://github.com/uber/cadence-idl/blob/3a2b613eaf9cebcd419e9926e09d02de9c7f15da/thrift/admin.thrift#L207

Thanks for spotting it, I have updated it to be int32, like everywhere else.